### ISSUE TITLE
Fish Pond recipe is out of date maybe?

### ISSUE THREAD

**Issue by Joyful-Attic** [Brian-Wuest/MC-Prefab#171]
1. What version of Minecraft, Forge and Mod are you running?
    1. Minecraft Version: 1.16.5
    1. Forge Version: 36.1.0
    1. Mod Version: 1.16.5 (Sorry if this is wrong, it's the only Version # I can find)
1. What type of issue is this? Choose one of the following:
    1. **Bug** (Something doesn't work the way you expect or the game crashes/has an issue)

1. If this is a **bug**, please provide as much detail as possible for the issue that you are running into. If possible post a log file to PasteBin or another similar site.

Sorry this is my first time doing this, I'm not very well oriented on asking about issues with a mod, so please forgive me if I post the incorrect log file.

I'd like to make the fish pond, however the recipe includes a fish icon that isn't from the latest update, (after aquatic update?) which I realized the page was last edited  Jul 29, 2016, so maybe the mod just isn't up to date. It seems you have a lot on your hands so I'm not demanding it immediately by any means, but I was wondering if it was possible to have it work with the latest update, maybe use a cod fish since its one of the more commonly/easily attainable fish?
I really enjoy your mod, I've only run into another issue where a recipe just won't work but I'm not tilted about it. I think it's great, I love using it, it gives me tasks to work towards accomplishing. 

![image](https://user-images.githubusercontent.com/84486543/118910890-4474c880-b8f3-11eb-9dd5-e1f003d9b181.png)

[latest.log](https://github.com/Brian-Wuest/MC-Prefab/files/6512478/latest.log)
[debug.log](https://github.com/Brian-Wuest/MC-Prefab/files/6512479/debug.log)

**Comment by Brian-Wuest** [comment]
Hmmm that's odd, it should be able to take all fish. I just haven't updated the image in a while. I will have to take a look, thank you!

**Comment by markygnlg** [comment]
To find the 'mod' version number you can either use the name of the file here [found in your 'mods' folder], ie: "prefab-1.x.x.xx.jar"

or you can goto the 'mods' button in the main menu & look for 'prefab' [you can sort your mods by 'A-Z' or 'Z-A' to find it easier]

I am assuming you're referring to a single player game though - I don't play on internet servers so I do not know how to find out 'mod' versions that way.

**Comment by markygnlg** [comment]
Also, "Joyful-Attic opened this issue 3 hours ago" but you stated "I realized the page was last edited Jul 29, 2016" - are you downloading this mod from curseforge? ie:

https://www.curseforge.com/minecraft/mc-mods/prefab

That is where all the files & updates are, AFAIK.

**EDIT**:
I just noticed the log files you included both of which mention v1.6.5.3 which is the current version.
The recipe uses a 'tag' which includes all fish of all mods that are labelled as a fish would work - do you use Just Enough Items (JEI) for your recipes?

https://www.curseforge.com/minecraft/mc-mods/jei

**Comment by Joyful-Attic** [comment]
> Also, "Joyful-Attic opened this issue 3 hours ago" but you stated "I realized the page was last edited Jul 29, 2016" - are you downloading this mod from curseforge? ie:
> 
> https://www.curseforge.com/minecraft/mc-mods/prefab
> 
> That is where all the files & updates are, AFAIK.
> 
> **EDIT**:
> I just noticed the log files you included both of which mention v1.6.5.3 which is the current version.
> The recipe uses a 'tag' which includes all fish of all mods that are labelled as a fish would work - do you use Just Enough Items (JEI) for your recipes?
> 
> https://www.curseforge.com/minecraft/mc-mods/jei

Oh okay, well I downloaded it from 9minecraft, which had said it was updated 3 weeks ago, mods folder says the jar file is 1.16.5
But I will give the curseforge version a try to see if it works!

And no I'm not using JEI

Edit: I was on this Github's page as I follow recipes since I have terrible memory and this is what I was referring to:
![2021-05-20 02_59_26-Window](https://user-images.githubusercontent.com/84486543/118933633-70a24080-b917-11eb-9fad-83c28bafd075.png)

**Comment by markygnlg** [comment]
TL:DR - put any fish/sapling into the recipe as those both use 'tags' as the required item, much like 'bundle of timber' can use any type of logs.


As a beginner to mods myself I can maybe help you, with reference to your comment "I'm not very well oriented on asking about issues with a mod". There are basically 2 types of recipe ingredients: items & tags (including 'forge tags').

An 'item' in a recipe is an item that is EXACTLY/SPECIFICALLY required whereas a 'tag' in a recipe is 'a wildcard/interchangable' such as any log vs a specific log; any planks/doors/slabs vs a specific planks/door/slab & any fish vs a specific fish (that's where JEI comes in handy).

**Comment by Brian-Wuest** [comment]
@Joyful-Attic - I didn't even remember that I had the GitHub wiki still around. I abandoned that when I got feedback on CurseForge that people there wanted to see the recipes/images/etc on CurseForge without having to go to another page or site.

**Comment by Brian-Wuest** [comment]
To answer your original question about the recipe; any fish you can catch with a fishing rod can be used in the recipe. In the screen shot below you can see that I have "Raw Salmon" in the recipe. Funnily enough; "Cooked Salmon" also works because of the Tagging system described above.

You will notice that a "Birch Sapling" is also in the recipe instead of an "Oak Sapling" because that part of the recipe also uses the Tagging system.

![image](https://user-images.githubusercontent.com/13502068/118963151-c84aa700-b92b-11eb-9c45-a54fe4a912f4.png)

**Comment by Joyful-Attic** [comment]
Sorry to make you wait so long, I had to sleep! 

So I downloaded the version from curseforge and instead of using sand which I think was the blocks used in the original recipe, used Brian-Wuest's screencap above (the double compressed dirt) and it worked! (I didn't have a birch sapling on me so I test out that 'tag' system with an oak sapling and it functioned all the same)
I might have just needed to update the version I had, or looked a little harder to make sure the version wasn't just 1.16.5 compatible and was actually really up to date.

Thank you for helping me! It means a lot!! 🙏 I'll have to use a mix of the curseforge site nad the github wiki for referring to recipes. I play in survival and I like building everything on my own. TYSVM!!

**Comment by markygnlg** [comment]
Good to hear!
The version 'problem' is why so many mod makers will only use curseforge. I'd seriously suggest JEI to get your recipes though - it's one of the most popular mc mods out there, like optifine/forge/fabric.

Many mod makers do not show recipes anymore because JEI is soooo heavily used ie: **111,879,784 Downloads** Last Updated: 4 days ago!!!
https://www.curseforge.com/minecraft/mc-mods/jei

Take care

**Comment by markygnlg** [comment]
Oh, & if it all works you can click on 'closed' on the bottom right to indicated that the 'issue' has been resolved. Happy gaming! & really, do get JEI!
